Lib/fontTools/subset/__init__.py

```diff
@@ -2829,6 +2815,24 @@ def _closure_glyphs(self, font):
 
 		self.reverseEmptiedGlyphMap = {g:order[g] for g in self.glyphs_emptied}
 
+		if not self.options.retain_gids:
+			new_glyph_order = [
+				g for g in glyph_order if g in self.glyphs_retained
+			]
+		else:
+			new_glyph_order = [
+				g for g in glyph_order
+				if font.getGlyphID(g) <= self.last_retained_order
+			]
```

Comment:
Optional, terser and arguably clearer if done with lambda and filter. Something like:

```
retention_fn = lambda g: g in self.glyphs_retained
if self.options.retain_gids:
  retention_fn = lambda g: font.getGlyphID(g) <= self.last_retained_order
new_glyph_order = list(filter(retention_fn, glyph_order))
```

Reply:
I simply moved that preexisting chunk of code